grammarinator/generate.py

```diff
@@ -286,10 +307,11 @@ def restricted_float(value):
                    transformers=args.transformer, serializer=args.serializer,
                    cleanup=False, encoding=args.encoding) as generator:
         if args.jobs > 1:
-            with Pool(args.jobs) as pool:
-                pool.map(generator, range(args.n))
+            shared_lock = Lock()
+            with Pool(args.jobs, initializer=generator.set_lock, initargs=(shared_lock,)) as pool:
+                pool.map(generator, count(0) if args.n == 'inf' else range(args.n))
```

Comment:
This line causes aggressive memory consumption leading to a crash if `inf` is set. @renatahodovan Do you have any idea what could be wrong?